### ISSUE TITLE
Find and linkify pincites

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#3401]
A user is running into an error today that the visualization tool doesn't show a citation from:

https://www.courtlistener.com/opinion/94422/seneca-nation-v-christy/

To: 

https://www.courtlistener.com/opinion/85404/johnson-grahams-lessee-v-mcintosh

The problem is that the citation looks like this: 

> And see Johnson v. McIntosh, 8 Wheat. 573

But the citation in _Johnson_ is actually `8 Wheat. 543`. So the citation in _Seneca Nation_ is to a pincite. 

I think in the past we've looked at these and sort of thrown up our hands because they're too difficult to figure out, particularly if we don't have every citation. I think we can do better now: 
 - We have code in the [citation lookup tool](https://www.courtlistener.com/c/) that can look up pincites
 - We have many, many more citations than we used to.

So I think at some point we should start thinking about this, particularly for older cases, where our citation collection is more complete.

**Comment by mlissner** [comment]
Yep. I think when we get to this we'll want to do some scoping work to think about whether to do it everywhere or just in some times or places to make it more reliable. 

It could also be that the whole citation is enough to make it work reliably. Like, this tells us a pincite and a case name. That might be enough:

> And see Johnson v. McIntosh, 8 Wheat. 573

**Comment by flooie** [comment]
You know- we already resolve pincites in the Citation page. 

For example [http://courtlistener.com/c/wheat/8/574/](http://courtlistener.com/c/wheat/8/574/) resovles naturally to [http://courtlistener.com/c/wheat/8/543/](http://courtlistener.com/c/wheat/8/543/) and eventually [https://www.courtlistener.com/opinion/85404/johnson-grahams-lessee-v-mcintosh/](https://www.courtlistener.com/opinion/85404/johnson-grahams-lessee-v-mcintosh/)

Why not do the same thing here.  if no match is found - resolve it the same way and just verify the name and date filed to match the opinion?

**Comment by quevon24** [comment]
@flooie Currently, when we resolve FullCaseCitations, we match them to opinions, not clusters. This is done using the function es_search_db_for_full_citation.

The problem arises when we have a possible match that contains multiple opinions. For example, consider this cluster:
https://www.courtlistener.com/opinion/108375/bivens-v-six-unknown-named-agents-of-federal-bureau-of-narcotics/

Using the es_search_db_for_full_citation function does not reduce the multiple results to a single one, which prevents some citations from being annotated.

This change requires a few updates in the code used for the find_citations command because it expects Opinions from the resolution, not an OpinionCluster.

I tried to determine how many changes I would need to make, and although there are several, it is possible to implement them.

The main issue lies in a small portion of the code responsible for creating Parenthetical and OpinionCited objects because both expect an opinion for the described_opinion_id and cited_opinion_id fields. Since we will now be returning clusters, we need to decide how to set these IDs if the matching cluster contains multiple opinions. You can see it here: https://github.com/freelawproject/courtlistener/blob/main/cl/citations/tasks.py#L216 and here: https://github.com/freelawproject/courtlistener/blob/main/cl/citations/tasks.py#L252

I was thinking that we could retrieve the opinions from the matching cluster, sort them by ID, and select the first one (the oldest) to use for creating the Parenthetical (if applicable) and OpinionCited objects.

These changes could also affect the recap_citations command because both commands use the do_resolve_citations function, which calls resolve_fullcase_citation, which in turn calls es_search_db_for_full_citation. Since this function also expects Opinions instead of Clusters, perhaps we could add an additional argument to the do_resolve_citations function to decide whether to call the regular resolve_fullcase_citation (for recap_citations), which returns opinions, or a new function that returns clusters (for find_citations).

**Comment by flooie** [comment]
we did this right?

**Comment by quevon24** [comment]
> we did this right?

Yes, have you finished running the `find_citations` command for NY to check that everything is ok?
